cedar-policy/src/api.rs

```diff
@@ -1169,21 +1383,38 @@ pub struct EntityTypeName(ast::Name);
 
 impl EntityTypeName {
     /// Get the basename of the `EntityTypeName` (ie, with namespaces stripped).
+    /// ```
+    /// use cedar_policy::EntityTypeName;
+    /// use std::str::FromStr;
+    /// let type_name = EntityTypeName::from_str("MySpace::User").unwrap();
+    /// assert_eq!(type_name.basename(), "User");
+    /// ```
     pub fn basename(&self) -> &str {
         self.0.basename().as_ref()
     }
 
     /// Get the namespace of the `EntityTypeName`, as components
+    /// ```
+    /// use cedar_policy::EntityTypeName;
+    /// use std::str::FromStr;
+    /// let type_name = EntityTypeName::from_str("Namespace::MySpace::User").unwrap();
+    /// let mut components = type_name.namespace_components();
+    /// assert_eq!(components.next(), Some("Namespace"));
+    /// assert_eq!(components.next(), Some("MySpace"));
+    /// assert_eq!(components.next(), None);
+    /// ```
     pub fn namespace_components(&self) -> impl Iterator<Item = &str> {
         self.0.namespace_components().map(AsRef::as_ref)
     }
 
     /// Get the full namespace of the `EntityTypeName`, as a single string.
-    ///
-    /// Examples:
-    /// - `foo::bar` --> the namespace is `"foo"`
-    /// - `bar` --> the namespace is `""`
-    /// - `foo::bar::baz` --> the namespace is `"foo::bar"`
+    /// ```
+    /// use cedar_policy::EntityTypeName;
+    /// use std::str::FromStr;
```

Comment:
let's hide these imports
```suggestion
    /// # use cedar_policy::EntityTypeName;
    /// # use std::str::FromStr;
```